plugin.video.embycon/resources/lib/simple_logging.py

```diff
@@ -0,0 +1,46 @@
+# Gnu General Public License - see LICENSE.TXT
+
+import xbmc
+import xbmcaddon
+
+
+class SimpleLogging():
+    level = 0
+    name = ""
+
+    def __init__(self, name):
+        settings = xbmcaddon.Addon(id='plugin.video.embycon')
+        prefix = settings.getAddonInfo('name')
+        log_level = settings.getSetting('logLevel')
+        self.level = int(log_level)
+        self.name = prefix + '.' + name
+
+    def getLevel(self):
+        return self.level
+
+    def __str__(self):
+        return "LogLevel: " + str(self.level)
+
+    def error(self, msg):
+        if (self.level >= 0):
+            try:
+                xbmc.log(self.format(msg, "ERROR"), level=xbmc.LOGNOTICE)
+            except UnicodeEncodeError:
+                xbmc.log(self.format(msg, "ERROR").encode('utf-8'), level=xbmc.LOGNOTICE)
+
+    def info(self, msg):
+        if (self.level >= 1):
+            try:
+                xbmc.log(self.format(msg, "INFO"), level=xbmc.LOGNOTICE)
+            except UnicodeEncodeError:
+                xbmc.log(self.format(msg, "INFO").encode('utf-8'), level=xbmc.LOGNOTICE)
+
+    def debug(self, msg):
+        if (self.level >= 2):
+            try:
+                xbmc.log(self.format(msg, "DEBUG"), level=xbmc.LOGNOTICE)
```

Comment:
Yep that is my point I guess, the addon currently has a log level setting, so if I need a user to submit a log I tell them to set the addon logging to debug and reproduce the issue and submit the log.
The model you are referring to is that the addon would not have any log level setting at all. It would just log with the desired log level and set the correct KODI log level in the xbmc.log call
This approach I am using is probably left over from Kodi 14 or even 13, this addon has been alive for a long time.
If I really need to change this I can I guess but as I said it will make my life harder, dealing with bigger log files was never fun and in fact when I see a user log submission with both the addon and kodi debug logging turned on I usually get them to re run with just the addon logging set to debug as trying to follow the large Kodi Debug log file is a pain.